From 256eda666b4687c5242b9f075caf4e09ad642daa Mon Sep 17 00:00:00 2001
From: Sivaprakash Murugesan <sivaprak@codeaurora.org>
Date: Fri, 17 Apr 2020 16:37:10 +0530
Subject: [PATCH] remoteproc: wcss: disable auto boot for IPQ8074

auto boot is disabled for IPQ8074 the wifi driver brings up the wcss.

Signed-off-by: Sivaprakash Murugesan <sivaprak@codeaurora.org>
Change-Id: Ia82edb7ee52f2bd010c099f151179d69a953ac88
---
 drivers/remoteproc/qcom_q6v5_wcss.c | 4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/remoteproc/qcom_q6v5_wcss.c
+++ b/drivers/remoteproc/qcom_q6v5_wcss.c
@@ -161,6 +161,7 @@ struct wcss_data {
 	const struct rproc_ops *ops;
 	bool requires_force_stop;
 	bool need_mem_protection;
+	bool need_auto_boot;
 };
 
 static int q6v5_wcss_reset(struct q6v5_wcss *wcss)
@@ -1147,6 +1148,7 @@ static int q6v5_wcss_probe(struct platfo
 						      desc->sysmon_name,
 						      desc->ssctl_id);
 
+	rproc->auto_boot = desc->need_auto_boot;
 	ret = rproc_add(rproc);
 	if (ret)
 		goto free_rproc;
@@ -1183,6 +1185,7 @@ static const struct wcss_data wcss_ipq80
 	.ops = &q6v5_wcss_ipq8074_ops,
 	.requires_force_stop = true,
 	.need_mem_protection = true,
+	.need_auto_boot = false,
 };
 
 static const struct wcss_data wcss_qcs404_res_init = {
@@ -1199,6 +1202,7 @@ static const struct wcss_data wcss_qcs40
 	.ssctl_id = 0x12,
 	.ops = &q6v5_wcss_qcs404_ops,
 	.requires_force_stop = false,
+	.need_auto_boot = true,
 };
 
 static const struct of_device_id q6v5_wcss_of_match[] = {
